src/llmcompressor/core/llmcompressor/train.py

```diff
@@ -0,0 +1,62 @@
+import math
+from typing import TYPE_CHECKING, Optional, Union
+
+from llmcompressor.args.training_arguments import TrainingArguments
+from llmcompressor.core import State
+from llmcompressor.core.llmcompressor.utils import LCDatasetArguments, parse_args
+from llmcompressor.datasets.utils import get_processed_dataset
+from llmcompressor.transformers.finetune.trainer import Trainer
+from llmcompressor.typing import DatasetType
+
+if TYPE_CHECKING:
+    from transformers.data.data_collator import DataCollator
+
+
+class HFSFTMixin:
+    state: State
+    train_dataset: Optional[DatasetType] = None
+    train_data_collator: Optional["DataCollator"] = None
+
+    def set_train_dataset(self, dataset: Union[str, DatasetType], **kwargs):
+        dataset_args = parse_args(LCDatasetArguments, dataset=dataset, **kwargs)
+
+        processed_dataset = get_processed_dataset(
+            dataset_args=dataset_args,
+            processor=self.state.processor,
+        )
+        self.train_dataset = processed_dataset.get("train")
+
+    def train(self, **kwargs):
+        raise NotImplementedError(
+            "Implementing LLMCompressor.train would require "
+            "changes which break existing training pathways"
+        )
```

Comment:
is this intentional?

Reply:
I called this out in the PR description, but implementing at this level would require major changes to existing pathways, which I leave as a todo for now